src/client.js

```diff
@@ -177,47 +152,81 @@ Client.prototype.connect = function(args) {
 		commands: args.commands
 	});
 
-	network.irc = irc;
-
 	client.networks.push(network);
 	client.emit("network", {
 		network: network
 	});
 
-	events.forEach(function(plugin) {
-		var path = "./plugins/irc-events/" + plugin;
-		require(path).apply(client, [
-			irc,
-			network
-		]);
-	});
+	var reconnect = function(args) {
+		network = _.find(client.networks, {id: network.id});
+
+		var stream = args.tls ? tls.connect(server) : net.connect(server);
 
-	irc.once("welcome", function() {
-		var delay = 1000;
-		var commands = args.commands;
-		if (Array.isArray(commands)) {
-			commands.forEach(function(cmd) {
-				setTimeout(function() {
-					client.input({
-						target: network.channels[0].id,
-						text: cmd
-					});
-				}, delay);
-				delay += 1000;
+		stream.on("error", function(e) {
+			console.log("Client#connect():\n" + e);
+			stream.end();
+			var msg = new Msg({
+				type: Msg.Type.ERROR,
+				text: "Connection error. Reconnecting in 30 seconds."
```

Comment:
The timeframe in this message doesn't match the setTimeout below -- maybe extract a variable?